### PR TITLE
[feat] Allow deprecation of fields in a future version and issue deprecation warnings also in prereleases

### PR DESCRIPTION
**Pull request by vkarak**
Also:

- Make unit tests for deprecation warnings more robust
-  Ignore prereleases when filtering deprecation warnings. For example, if we are on 3.5.0-dev.0 and a feature is set to deprecate in 3.5.0, then we issue the deprecation warning also for the prerelease.
- Fix deprecation warnings introduced by #1761.

This is a follow up of #1761.

### CODE REVIEW

**Review by ekouts — APPROVED** [review]

File: unittests/test_warnings.py [hunk 1]
```diff
@@ -23,7 +23,9 @@ def test_deprecation_warning():
 
 def test_deprecation_warning_from_version():
```
Comment (teojgo) [inline unittests/test_warnings.py:24]: Is it possible to test a case where the a deprecation warning is thrown. Like test for a version previous to this one.

**Review by teojgo — CHANGES_REQUESTED** [review]

File: unittests/test_warnings.py (hunk 1 above)
Comment (vkarak) [reply unittests/test_warnings.py:24]: The default `from_version` is `0.0.0`, so what you ask here is implicitly tested, isn't it?

Comment (teojgo) [reply unittests/test_warnings.py:24]: I mean a case where the deprecation warning is actually thrown.

Comment (vkarak) [reply unittests/test_warnings.py:24]: I think the `test_deprecation_warning` does already this.

**Review by teojgo — APPROVED** [review]

**Pull request merged**